src/assets/js/battle.js

```diff
@@ -0,0 +1,183 @@
+    const enemyHTML = document.querySelector('#enemyArticle').children;
+    console.log(randomNumber);
+    if (randomNumber === 1) {
+      populateEnemyData(enemyHTML, goblin);
+    } else if (randomNumber === 2) {
+      populateEnemyData(enemyHTML, elf);
+    } else if (randomNumber === 3) {
+      populateEnemyData(enemyHTML, giant);
+    } else {
+      populateEnemyData(enemyHTML, rat);
+    }
+  };
+  // Function to populate the enemy data to be called elsewhere
+  const populatePlayerData = (tag, play) => {
+    // Populate the name
+    tag[0].innerHTML = `<i class="fas fa-user mx-1"></i>${play.name}`;
+    // Populate the Current HP
+    tag[1].innerHTML = `<i class="fas fa-heart mx-1"></i>${play.currentHP}`;
+    // Populate the Weapon of Choice
+    tag[2].innerHTML = `<i class="fas fa-fire mx-1"></i>${play.weapon.weaponName}`;
+    // Populate the Armor Amount
+    tag[3].innerHTML = `<i class="fas fa-shield-alt mx-1"></i>${play.defense}`;
+    // Populate the Gold Amount
+    tag[4].innerHTML = `<i class="fas fa-money-bill-alt mx-1"></i>${play.gold}`;
+  };
+  // Function to populate the enemy data to be called elsewhere
+  const populateEnemyData = (tag, enemy) => {
+    // Populate the enemy name
+    tag[0].innerHTML = `<i class="fas fa-user mx-1"></i>${enemy.name}`;
+    // Populate the enemy HP
+    tag[1].innerHTML = `<i class="fas fa-heart mx-1"></i>${enemy.currentHP}`;
+    localStorage.setItem('currentEnemy', enemy.name);
+  };
+  // Function to determine which enemy is being fought
+  const attackEnemy = (play, gob, el, gi, ra) => {
+    const currentEnemy = localStorage.getItem('currentEnemy');
+    // Check the current enemy to determine the attacker
+    if (currentEnemy.toLowerCase() === 'goblin') {
+      attackAmount(play, gob);
+      populateEnemyData(document.querySelector('#enemyArticle').children, gob);
+    } else if (currentEnemy.toLowerCase() === 'elf') {
+      attackAmount(play, el);
+      populateEnemyData(document.querySelector('#enemyArticle').children, el);
+    } else if (currentEnemy.toLowerCase() === 'giant') {
+      attackAmount(play, gi);
+      populateEnemyData(document.querySelector('#enemyArticle').children, gi);
+    } else if (currentEnemy.toLowerCase() === 'rat') {
+      attackAmount(play, ra);
+      populateEnemyData(document.querySelector('#enemyArticle').children, ra);
+    }
+    // Check if the enemy or player has died and call the enemy decider function
+    gob.currentHP <= 0 ? enemyDecider(play, gob, el, gi, ra) : null;
+    el.currentHP <= 0 ? enemyDecider(play, gob, el, gi, ra) : null;
+    gi.currentHP <= 0 ? enemyDecider(play, gob, el, gi, ra) : null;
+    ra.currentHP <= 0 ? enemyDecider(play, gob, el, gi, ra) : null;
+    play.currentHP <= 0 ? deadPlayer(play) : null;
+    console.log(currentEnemy);
+  };
+  // Function to determine the attack strength
+  const attackAmount = (play, enemy) => {
+    // Random number and DOM element
```

Comment:
Use meaningful comments, adding unneeded comments can make the code a pain to read. [link to info on comments in javascript](https://www.w3.org/wiki/JavaScript_best_practices#Comment_as_much_as_needed_but_not_more)

Reply:
I'll definitely take a look at that. Thanks!